rpc-test/tests/rpc.rs

```diff
@@ -325,28 +337,21 @@ fn test_rpc_subscriptions() {
                         .await
                         .unwrap();
 
+                    account_subscription_ready_clone_clone.fetch_add(1, Ordering::SeqCst);
+
                     let response = account_notifications.next().await.unwrap();
-                    account_sender.send(response).unwrap();
+                    account_sender.send((pubkey, response)).unwrap();
                     account_unsubscribe().await;
                 }
             });
         }
-
-        // Signal ready after the next slot notification
-        tokio::spawn({
-            let _pubsub_client = Arc::clone(&pubsub_client);
-            async move {
-                let (mut slot_notifications, slot_unsubscribe) =
-                    _pubsub_client.slot_subscribe().await.unwrap();
-                let _response = slot_notifications.next().await.unwrap();
-                ready_sender.send(()).unwrap();
-                slot_unsubscribe().await;
-            }
-        });
     });
 
-    // Wait for signature subscriptions
-    ready_receiver.recv_timeout(Duration::from_secs(2)).unwrap();
+    while signature_subscription_ready.load(Ordering::SeqCst) != transactions.len()
+        || account_subscription_ready.load(Ordering::SeqCst) != transactions.len()
+    {
+        sleep(Duration::from_millis(100))
```

Comment:
Can you please add some sort of timeout to this loop so that it doesn't hang forever if one of the `AtomicUsize`s doesn't reach `transactions.len()` for some reason?

Reply:
right! I gave 15 secs countdown and then checked both of them whether been setup properly. the commit [4fe52cd](https://github.com/solana-labs/solana/pull/27394/commits/4fe52cdd2ea776219f7d13ee42e9fcc956fac917)